### PR TITLE
Add 3D app icon setting for UWP

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

Adds the ability to set a 3D app launcher model for WMR UWP.
https://docs.microsoft.com/en-us/windows/mixed-reality/distribute/creating-3d-models-for-use-in-the-windows-mixed-reality-home

The settings can be found under Project Settings -> MRTK3 -> Build Settings

image

Works great with #10923 🥳

image

## Changes

- Fixes: #10929

### CODE REVIEW

File: com.microsoft.mrtk.core/Editor/MRTKBuildPreferences.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        private static GUIContent buttonContent = null;
+        private static UnityEditor.Editor gameObjectEditor = null;
+        private static GUIStyle appLauncherPreviewBackgroundColor = null;
+        private static bool isBuilding = false;
+
+        // Arbitrary callback order, chosen to be larger so that it runs after other things that
+        // a developer may have already.
+        int IOrderedCallback.callbackOrder => 100;
+
+        /// <remarks>See <see href="https://docs.microsoft.com/windows/mixed-reality/distribute/3d-app-launcher-design-guidance">3D app launcher design guidance</see> for more information.</remarks>
```
Comment (davidkline-ms) [inline com.microsoft.mrtk.core/Editor/MRTKBuildPreferences.cs:34]: I have not seen remarks doc comments w/o a summary before.... should we add one here? (i know it is private, but.... doesnt feel right)

Comment (keveleigh) [reply com.microsoft.mrtk.core/Editor/MRTKBuildPreferences.cs:34]: I mostly just wanted this info _somewhere_. I can adjust the location

**Review by davidkline-ms — DISMISSED** [review]
Approved with the one question re: doc comments

File: com.microsoft.mrtk.core/Editor/MRTKBuildPreferences.cs (hunk 1 above)
Comment (davidkline-ms) [reply com.microsoft.mrtk.core/Editor/MRTKBuildPreferences.cs:34]: then feel free to merge, my friend :)

Comment (keveleigh) [reply com.microsoft.mrtk.core/Editor/MRTKBuildPreferences.cs]: Moved this to the class summary!

**Review by MaxWang-MS — APPROVED** [review]

**Pull request merged**